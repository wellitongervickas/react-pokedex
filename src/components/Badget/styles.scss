@import '../../assets/scss/variables';

.badget {
  display: inline-block;
  border: 1px solid $gray-light;
  border-radius: $size-default;
  padding: $size-small4 $size-small2;
}
