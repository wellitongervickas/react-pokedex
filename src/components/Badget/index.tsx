import React, { ReactNode } from 'react';

import 'components/Badget/styles.scss';

interface BadgetInterface {
  children: ReactNode,
};

const Badget = ({ children }: BadgetInterface) => (
  <div className="badget">{children}</div>
);

export default Badget;
