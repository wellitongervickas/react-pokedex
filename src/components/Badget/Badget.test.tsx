import React from 'react';
import { render } from '@testing-library/react';

import Badget from '.';

it('renders badget component', () => {
  const { getByText, container } = render(<Badget>Hello World</Badget>);

  expect(getByText('Hello World')).toBeInTheDocument();
  expect(container.firstElementChild?.classList.contains('badget')).toBeTruthy();
  expect(container.firstElementChild).toMatchSnapshot();
});
