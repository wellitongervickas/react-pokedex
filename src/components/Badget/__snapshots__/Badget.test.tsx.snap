// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders badget component 1`] = `
<div
  class="badget"
>
  Hello World
</div>
`;
